flow-server/src/test/java/com/vaadin/flow/server/communication/StreamRequestHandlerTest.java

```diff
@@ -143,6 +146,44 @@ public void stateNodeStates_handlerMustNotReplyWhenNodeDisabled()
         Mockito.verify(response).sendError(403, "Resource not available");
     }
 
+    @Test
+    public void stateNodeStates_handlerMustReplyWhenNodeDisabledButAllowsAllMode()
+            throws IOException {
+        TestElementHandlerBuilder builder = new TestElementHandlerBuilder()
+                .withDisabledUpdateMode(DisabledUpdateMode.ALWAYS);
+        stateNodeStatesTestInternal(builder);
+        Mockito.verify(response, Mockito.never()).sendError(Mockito.anyInt(),
+                Mockito.anyString());
+    }
+
+    @Test
+    public void stateNodeStates_handlerMustNotReplyWhenNodeInert()
+            throws IOException {
+        TestElementHandlerBuilder builder = new TestElementHandlerBuilder()
+                .withInert(true);
+        stateNodeStatesTestInternal(builder);
+        Mockito.verify(response).sendError(403, "Resource not available");
+    }
+
+    @Test
+    public void stateNodeStates_handlerMustReplyWhenNodeInertButAllowsInert()
+            throws IOException {
+        TestElementHandlerBuilder builder = new TestElementHandlerBuilder()
+                .withInert(true).withAllowInert(true);
+        stateNodeStatesTestInternal(builder);
+        Mockito.verify(response, Mockito.never()).sendError(Mockito.anyInt(),
+                Mockito.anyString());
+    }
+
+    @Test
+    public void stateNodeStates_handlerMustNotReplyWhenNodeHidden()
```

Comment:
Test name is wrong as the expectation is that there is the response 403.

```suggestion
    public void nodeHidden_shouldRespondWithResourceNotAvailable()
```